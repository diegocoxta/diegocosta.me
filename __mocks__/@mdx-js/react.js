module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
