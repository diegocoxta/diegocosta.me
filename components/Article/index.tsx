import { MDXRemote } from 'next-mdx-remote/rsc';

import type { BlogContentAttributes } from '~/app/cms';

import CodeBlock from '~/components/CodeBlock';

import styles from './styles.module.css';

export default function Article(props: { children: BlogContentAttributes['content'] | undefined }): React.ReactElement {
  return (
    <div className={styles.article}>
      {props.children && (
        <MDXRemote
          source={props.children}
          components={{
            code: CodeBlock,
          }}
        />
      )}
    </div>
  );
}
