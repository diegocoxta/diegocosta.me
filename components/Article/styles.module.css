.article {
  line-height: 1.5;
  font-size: 19px;
  margin: 0 auto 60px;
  font-weight: 400;

  :global(.responsive) {
    position: relative;
    padding-bottom: 56.25%;
    height: 0;
    overflow: hidden;
    max-width: 100%;
  }

  :global(.responsive iframe),
  :global(.responsive object),
  :global(.responsive embed) {
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
  }

  a {
    color: var(--main-accent-color);
    text-decoration: none;
    border-bottom: 1px solid var(--main-background-color);

    &:hover,
    &:focus {
      border-bottom: 1px solid var(--main-accent-color);
      outline: none;
    }
  }

  iframe {
    width: 100%;
    border: 0;
  }

  blockquote {
    border-left: 5px solid var(--main-accent-color);
    padding-left: 20px;
    margin-left: 0;
  }

  img {
    display: block;
    margin: 0 auto;
  }

  h3 {
    font-size: 28px;
  }
}
