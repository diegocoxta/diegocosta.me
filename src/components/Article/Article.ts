import styled from 'styled-components';

export default styled.article`
  margin: 0 0 60px 0;
  font-weight: 400;
  color: ${({ theme }) => theme.textColor};
`;

export const Content = styled.section`
  line-height: 1.5;

  a {
    color: ${({ theme }) => theme.accentColor};
    text-decoration: none;
    border-bottom: 1px solid ${({ theme }) => theme.backgroundColor};

    :hover,
    :focus {
      border-bottom: 1px solid ${({ theme }) => theme.accentColor};
      outline: none;
    }
  }

  blockquote {
    border-left: 5px solid ${({ theme }) => theme.accentColor};
    padding-left: 20px;
  }

  .gatsby-highlight {
    padding: 10px 0;
  }

  pre[class*='language-'],
  code[class*='language-'],
  .gatsby-highlight {
    background-color: ${({ theme }) => theme.backgroundColor};
    font-family: monospace;
  }

  .line-numbers-rows,
  .line-numbers-rows span::before {
    border: 0.5px solid ${({ theme }) => theme.backgroundColor};
  }

  img {
    box-shadow: none !important;
  }
`;
