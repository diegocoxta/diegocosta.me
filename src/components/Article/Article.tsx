import React from 'react';
import styled from 'styled-components';
import kebabCase from 'lodash.kebabcase';
import { MDXRenderer } from 'gatsby-plugin-mdx';

import { usei18n, Link } from '~/utils/i18n';

import FixedContainer from '~/components/FixedContainer';

const Content = styled.article`
  margin: 0 0 60px 0;
  font-weight: 400;
  color: ${({ theme }) => theme.textColor};
`;

const Header = styled.header`
  margin: 20px 0 0 0;
`;

const Title = styled.h2`
  font-size: 36px;
  margin: 0;
`;

const CustomLink = styled(Link)`
  color: ${({ theme }) => theme.titleColor};
  box-shadow: none;
  text-decoration: none;

  :hover,
  :focus {
    border-bottom: 1px solid ${({ theme }) => theme.titleColor};
    outline: none;
  }
`;

const TagLink = styled(Link)`
  text-decoration: none;
  box-shadow: none;
  font-size: 14px;
  color: ${({ theme }) => theme.accentColor};
  font-weight: 700;
  text-transform: lowercase;

  :hover,
  :focus {
    border-bottom: 1px solid ${({ theme }) => theme.accentColor};
    outline: none;
  }
`;

const TagList = styled.ul`
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 0;
  padding: 0;
`;

const TagItem = styled.li`
  list-style: none;
  padding: 0 10px 0 0;
  margin-bottom: 5px;
`;

const Details = styled.p`
  color: ${({ theme }) => theme.subtitleColor};
  font-size: 18px;
  margin: 5px 0;
`;

export const Body = styled.section`
  line-height: 1.5;

  a {
    color: ${({ theme }) => theme.accentColor};
    text-decoration: none;
    border-bottom: 1px solid ${({ theme }) => theme.backgroundColor};

    :hover,
    :focus {
      border-bottom: 1px solid ${({ theme }) => theme.accentColor};
      outline: none;
    }
  }

  blockquote {
    border-left: 5px solid ${({ theme }) => theme.accentColor};
    padding-left: 20px;
  }

  .gatsby-highlight {
    padding: 10px 0;
  }

  pre[class*='language-'],
  code[class*='language-'],
  .gatsby-highlight {
    background-color: ${({ theme }) => theme.backgroundColor};
    font-family: monospace;
    white-space: pre-wrap;
  }

  .line-numbers-rows,
  .line-numbers-rows span::before {
    border: 0.5px solid ${({ theme }) => theme.backgroundColor};
  }

  img {
    box-shadow: none !important;
  }

  h3 {
    font-size: 28px;
  }
`;

export interface ArticleProps {
  title: string;
  url?: string | null;
  readingTime?: number;
  language?: string | null;
  date?: string;
  tags?: string[] | null;
  description?: string | null;
  mdxContent?: string | null;
  showArticleDetails?: boolean;
}

export default function Article(props: ArticleProps): React.ReactElement {
  const i18n = usei18n();

  const pageLanguage = i18n.getCurrentLanguage();

  const getReadingTime = () => {
    if (!props.readingTime) {
      return undefined;
    }

    const lessThan1Minute = i18n.getTranslationFor('article.lessThan1Minute');
    const ofReading = i18n.getTranslationFor('article.ofReading');
    const minutes = i18n.getTranslationFor('article.minutes');

    if (props.readingTime < 1) {
      return ` · ${lessThan1Minute} ${ofReading}`;
    }

    return ` · ${props.readingTime?.toFixed()} ${minutes} ${ofReading}`;
  };

  const date =
    props.date &&
    new Date(props.date).toLocaleDateString(pageLanguage, {
      year: 'numeric',
      month: '2-digit',
      day: '2-digit',
    });

  const language = () => {
    if (!props.language) {
      return;
    }

    const languagePrefix = i18n.getTranslationFor('article.languagePrefix');
    const languageName = i18n.getTranslationFor(`languages.${props.language}`);

    return ` · ${languagePrefix} ${languageName}`;
  };

  const Tags = props.showArticleDetails && props.tags && (
    <TagList data-testid="article-header-tags">
      {props.tags.map((tag: string, index: number) => (
        <TagItem key={`${index}-${tag}`} data-testid="article-header-tag">
          <TagLink to={`/tags/${kebabCase(tag ?? '')}`}>#{kebabCase(tag ?? '')}</TagLink>
        </TagItem>
      ))}
    </TagList>
  );

  return (
    <FixedContainer>
      <Content data-testid="article-item">
        <Header>
          <Title>
            {props.url ? (
              <CustomLink to={props.url} data-testid="article-header-custom-link" language={props.language}>
                {props.title}
              </CustomLink>
            ) : (
              props.title
            )}
          </Title>
          {props.showArticleDetails && (
            <Details>
              {date} {getReadingTime()} {language()}
            </Details>
          )}
          {Tags}
        </Header>
        {props.description && <Body dangerouslySetInnerHTML={{ __html: props.description }} />}
        {props.mdxContent && (
          <Body>
            <MDXRenderer>{props.mdxContent}</MDXRenderer>
          </Body>
        )}
      </Content>
    </FixedContainer>
  );
}

Article.defaultProps = {
  showArticleDetails: true,
};
