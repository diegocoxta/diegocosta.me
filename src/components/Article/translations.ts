export type Translations = typeof en;

const en = {
  lessThan1Minute: 'Less than 1 minute',
  minutes: 'minutes',
  ofReading: 'of reading',
};

const pt: Translations = {
  lessThan1Minute: 'Menos de 1 minuto',
  minutes: 'minutos',
  ofReading: 'de leitura',
};

const translations: { [key: string]: Translations } = {
  pt,
  en,
};

export default translations;
