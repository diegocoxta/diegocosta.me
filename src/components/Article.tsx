import React from 'react';
import styled from 'styled-components';
import { graphql } from 'gatsby';

import { useLocale, getContentLanguage, Link } from '@app/components/LanguageSwitcher';

const Container = styled.article`
  margin: 0 0 60px 0;
  font-weight: 400;
  color: ${({ theme }) => theme.textColor};
`;

const Header = styled.header`
  margin: 20px 0 0 0;
`;

const MetaAttributes = styled.p`
  color: ${({ theme }) => theme.subtitleColor};
  font-size: 19px;
  margin: 10px 0;
`;

export const Content = styled.section`
  line-height: 1.5;
  font-size: 19px;
  margin-top: 10px;

  a {
    color: ${({ theme }) => theme.accentColor};
    text-decoration: none;
    border-bottom: 1px solid ${({ theme }) => theme.backgroundColor};

    &:hover,
    &:focus {
      border-bottom: 1px solid ${({ theme }) => theme.accentColor};
      outline: none;
    }
  }

  code {
    background: ${({ theme }) => theme.accentColor}40;
    padding: 2px 6px;
    border-radius: 10px;
    font-size: 16px;
  }

  blockquote {
    border-left: 5px solid ${({ theme }) => theme.accentColor};
    padding-left: 20px;
    margin-left: 0;
  }

  img {
    box-shadow: none !important;
  }

  h3 {
    font-size: 28px;
  }

  table.uses-table {
    width: 100%;
    margin: 40px 0;

    td {
      text-align: center;
    }
  }
`;

const TagList = styled.ul`
  display: flex;
  flex-wrap: wrap;
  padding: 0;
  margin: 0;
`;

const TagItem = styled.li`
  list-style: none;
  padding: 0 10px 0 0;
`;

const TagLink = styled(Link)`
  text-decoration: none;
  box-shadow: none;
  font-size: 16px;
  color: ${({ theme }) => theme.accentColor};
  font-weight: 700;
  text-transform: lowercase;

  &:hover,
  &:focus {
    border-bottom: 1px solid ${({ theme }) => theme.accentColor};
    outline: none;
  }
`;

const Title = styled(Link).attrs((props) => ({
  as: props.to ? Link : 'h2',
}))`
  color: ${({ theme }) => theme.titleColor};
  box-shadow: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: 700;

  &:hover,
  &:focus {
    border-bottom: ${({ theme, to }) => to && `1px solid ${theme.titleColor}`};
    outline: none;
  }
`;

type Nullable<T> = T | null;

export type ArticleProps = {
  readingTime?: number;
  language?: Nullable<string>;
  date?: Nullable<string>;
  content?: Nullable<string>;
  kind?: Nullable<string>;
  tags?: string[] | null;
  title: string;
  url?: string | null;
  showContent?: boolean | null;
};

export default function Article(props: ArticleProps): React.ReactElement {
  const locale = useLocale();

  const pageLanguage = locale.getCurrentLanguage();

  const getReadingTime = () => {
    if (!props.readingTime) {
      return undefined;
    }

    const lessThan1Minute = locale.getTranslationFor('article.lessThan1Minute');
    const ofReading = locale.getTranslationFor('article.ofReading');
    const minutes = locale.getTranslationFor('article.minutes');

    if (props.readingTime < 1) {
      return `· ${lessThan1Minute} ${ofReading}`;
    }

    return `· ${props.readingTime?.toFixed()} ${minutes} ${ofReading}`;
  };

  const date =
    props.date &&
    new Date(props.date).toLocaleDateString(pageLanguage, {
      year: 'numeric',
      month: '2-digit',
      day: '2-digit',
    });

  return (
    <Container data-testid="article-item">
      <Header>
        <Title to={props.url} data-testid="article-header-title" language={props.url ? props.language : undefined}>
          {props.title}
        </Title>
        {props.kind === 'articles' && (
          <>
            <MetaAttributes>
              {date} {getReadingTime()} · {getContentLanguage(locale, props.language)}
            </MetaAttributes>
            <TagList data-testid="article-header-tags">
              {props.tags?.map((tag: string, index: number) => (
                <TagItem key={`${index}-${tag}`} data-testid="article-header-tag">
                  <TagLink to={`/tags/${tag}`}>{`#${tag}`}</TagLink>
                </TagItem>
              ))}
            </TagList>
          </>
        )}
      </Header>
      {props.content && <Content dangerouslySetInnerHTML={{ __html: props.content }} />}
    </Container>
  );
}

export const query = graphql`
  fragment ArticleInformation on MarkdownRemark {
    html
    excerpt
    fields {
      collection
      slug
      language
      readingTime {
        minutes
      }
    }
    frontmatter {
      date
      title
      description
      tags
      flags
    }
  }
`;
