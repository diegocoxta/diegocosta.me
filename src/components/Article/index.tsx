import React from 'react';
import { graphql } from 'gatsby';

import Article from './Article';

interface Props {
  article?: Queries.ArticleInformationFragment | null;
  showContent?: boolean | null;
}

export default (props: Props): React.ReactElement => {
  const { article, showContent } = props;
  const { frontmatter, fields, html, excerpt } = article ?? {};

  return (
    <Article
      kind={fields?.collection}
      title={frontmatter?.title ?? ''}
      date={frontmatter?.date}
      url={fields?.slug}
      tags={frontmatter?.tags as string[]}
      readingTime={fields?.readingTime?.minutes ?? 0}
      language={fields?.language}
      content={showContent ? html : frontmatter?.description || excerpt}
    />
  );
};

export const query = graphql`
  fragment ArticleInformation on MarkdownRemark {
    html
    excerpt
    fields {
      collection
      slug
      language
      readingTime {
        minutes
      }
    }
    frontmatter {
      date
      title
      description
      tags
      homepage_view_full_article
      status
    }
  }
`;
