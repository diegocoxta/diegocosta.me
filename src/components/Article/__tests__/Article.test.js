import React from 'react';
import { render } from '@testing-library/react';

import Article from '../Article';

describe('<Article />', () => {
  it('should render properly', () => {
    const { baseElement, getByTestId, getByText, getAllByTestId } = render(
      <Article
        title="Awesome Article"
        url="/awesome-article"
        readingTime={5}
        language="en"
        date="20/07/2020"
        tags={['jest', 'testing-library']}
      />
    );
    expect(getByText('Awesome Article')).toBeTruthy();
    expect(getByText('20/07/2020 · 5 minutes of reading')).toBeTruthy();
    expect(getByTestId('article-header-custom-link').href).toBe('http://localhost/awesome-article');
    expect(getByTestId('article-header-tags')).toBeTruthy();
    expect(getAllByTestId('article-header-tag').length).toEqual(2);
    expect(baseElement).toMatchSnapshot();
  });

  it('should not render the tags', () => {
    const { baseElement, queryByTestId, queryAllByTestId, getByText } = render(
      <Article title="Awesome Article" url="/awesome-article" readingTime={5} language="pt" date="20/07/2020" />
    );

    expect(queryByTestId('article-header-tags')).toBeFalsy();
    expect(getByText('20/07/2020 · 5 minutos de leitura')).toBeTruthy();
    expect(queryAllByTestId('article-header-tag').length).toEqual(0);
    expect(baseElement).toMatchSnapshot();
  });

  it('should not render the link', () => {
    const { baseElement, queryByTestId } = render(
      <Article title="Awesome Article" readingTime={5} language="en" date="20/07/2020" />
    );

    expect(queryByTestId('article-header-custom-link')).toBeFalsy();
    expect(baseElement).toMatchSnapshot();
  });

  it('should render properly in Poruguese', () => {
    const { baseElement, getByText } = render(
      <Article title="Awesome Article" url="/awesome-article" readingTime={5} language="pt" date="20/07/2020" />
    );

    expect(getByText('20/07/2020 · 5 minutos de leitura')).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
