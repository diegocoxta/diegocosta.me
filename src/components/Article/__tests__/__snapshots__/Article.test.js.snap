// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Article /> does not render the article details 1`] = `
.c2 {
  box-shadow: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: 700;
}

.c2:hover,
.c2:focus {
  border-bottom: 1px solid undefined;
  outline: none;
}

.c0 {
  margin: 0 0 60px 0;
  font-weight: 400;
}

.c1 {
  margin: 20px 0 0 0;
}

<body>
  <div>
    <article
      class="c0"
      data-testid="article-item"
    >
      <header
        class="c1"
      >
        <a
          class="c2"
          data-testid="article-header-title"
          href="/awesome-article"
          language="en"
        >
          Awesome Article
        </a>
      </header>
    </article>
  </div>
</body>
`;

exports[`<Article /> does not render the article language name 1`] = `
.c2 {
  box-shadow: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: 700;
}

.c2:hover,
.c2:focus {
  outline: none;
}

.c0 {
  margin: 0 0 60px 0;
  font-weight: 400;
}

.c1 {
  margin: 20px 0 0 0;
}

<body>
  <div>
    <article
      class="c0"
      data-testid="article-item"
    >
      <header
        class="c1"
      >
        <h2
          class="c2"
          data-testid="article-header-title"
        >
          Awesome Article
        </h2>
      </header>
    </article>
  </div>
</body>
`;

exports[`<Article /> does not render the link 1`] = `
.c2 {
  box-shadow: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: 700;
}

.c2:hover,
.c2:focus {
  outline: none;
}

.c0 {
  margin: 0 0 60px 0;
  font-weight: 400;
}

.c1 {
  margin: 20px 0 0 0;
}

<body>
  <div>
    <article
      class="c0"
      data-testid="article-item"
    >
      <header
        class="c1"
      >
        <h2
          class="c2"
          data-testid="article-header-title"
        >
          Awesome Article
        </h2>
      </header>
    </article>
  </div>
</body>
`;

exports[`<Article /> does not render the reading time 1`] = `
.c2 {
  box-shadow: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: 700;
}

.c2:hover,
.c2:focus {
  outline: none;
}

.c4 {
  display: flex;
  flex-wrap: wrap;
  padding: 0;
  margin: 0;
}

.c0 {
  margin: 0 0 60px 0;
  font-weight: 400;
}

.c1 {
  margin: 20px 0 0 0;
}

.c3 {
  font-size: 19px;
  margin: 10px 0;
}

<body>
  <div>
    <article
      class="c0"
      data-testid="article-item"
    >
      <header
        class="c1"
      >
        <h2
          class="c2"
          data-testid="article-header-title"
        >
          Awesome Article
        </h2>
        <p
          class="c3"
        >
          18/01/2020
           
           · 
          article.languagePrefix languages.en
        </p>
        <ul
          class="c4"
          data-testid="article-header-tags"
        />
      </header>
    </article>
  </div>
</body>
`;

exports[`<Article /> does not render the tags 1`] = `
.c2 {
  box-shadow: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: 700;
}

.c2:hover,
.c2:focus {
  border-bottom: 1px solid undefined;
  outline: none;
}

.c0 {
  margin: 0 0 60px 0;
  font-weight: 400;
}

.c1 {
  margin: 20px 0 0 0;
}

<body>
  <div>
    <article
      class="c0"
      data-testid="article-item"
    >
      <header
        class="c1"
      >
        <a
          class="c2"
          data-testid="article-header-title"
          href="/awesome-article"
          language="pt"
        >
          Awesome Article
        </a>
      </header>
    </article>
  </div>
</body>
`;

exports[`<Article /> renders properly 1`] = `
.c2 {
  box-shadow: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: 700;
}

.c2:hover,
.c2:focus {
  border-bottom: 1px solid undefined;
  outline: none;
}

.c4 {
  display: flex;
  flex-wrap: wrap;
  padding: 0;
  margin: 0;
}

.c5 {
  list-style: none;
  padding: 0 10px 0 0;
}

.c6 {
  text-decoration: none;
  box-shadow: none;
  font-size: 16px;
  font-weight: 700;
  text-transform: lowercase;
}

.c6:hover,
.c6:focus {
  border-bottom: 1px solid;
  outline: none;
}

.c0 {
  margin: 0 0 60px 0;
  font-weight: 400;
}

.c1 {
  margin: 20px 0 0 0;
}

.c3 {
  font-size: 19px;
  margin: 10px 0;
}

<body>
  <div>
    <article
      class="c0"
      data-testid="article-item"
    >
      <header
        class="c1"
      >
        <a
          class="c2"
          data-testid="article-header-title"
          href="/awesome-article"
          language="en"
        >
          Awesome Article
        </a>
        <p
          class="c3"
        >
          18/01/2020
           
          · 5 article.minutes article.ofReading
           · 
          article.languagePrefix languages.en
        </p>
        <ul
          class="c4"
          data-testid="article-header-tags"
        >
          <li
            class="c5"
            data-testid="article-header-tag"
          >
            <a
              class="c6"
              href="/tags/jest"
            >
              #jest
            </a>
          </li>
          <li
            class="c5"
            data-testid="article-header-tag"
          >
            <a
              class="c6"
              href="/tags/testing-library"
            >
              #testing-library
            </a>
          </li>
        </ul>
      </header>
    </article>
  </div>
</body>
`;

exports[`<Article /> renders properly the body content 1`] = `
.c2 {
  box-shadow: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: 700;
}

.c2:hover,
.c2:focus {
  border-bottom: 1px solid undefined;
  outline: none;
}

.c0 {
  margin: 0 0 60px 0;
  font-weight: 400;
}

.c1 {
  margin: 20px 0 0 0;
}

.c3 {
  line-height: 1.5;
  font-size: 19px;
  margin-top: 10px;
}

.c3 a {
  text-decoration: none;
  border-bottom: 1px solid;
}

.c3 a:hover,
.c3 a:focus {
  border-bottom: 1px solid;
  outline: none;
}

.c3 code {
  background: 40;
  padding: 2px 6px;
  border-radius: 10px;
  font-size: 16px;
}

.c3 blockquote {
  border-left: 5px solid;
  padding-left: 20px;
  margin-left: 0;
}

.c3 img {
  box-shadow: none!important;
}

.c3 h3 {
  font-size: 28px;
}

.c3 table.uses-table {
  width: 100%;
  margin: 40px 0;
}

.c3 table.uses-table td {
  text-align: center;
}

<body>
  <div>
    <article
      class="c0"
      data-testid="article-item"
    >
      <header
        class="c1"
      >
        <a
          class="c2"
          data-testid="article-header-title"
          href="/awesome-article"
          language="pt"
        >
          Awesome Article
        </a>
      </header>
      <section
        class="c3"
      >
        Article as a Children
      </section>
    </article>
  </div>
</body>
`;

exports[`<Article /> renders properly the description 1`] = `
.c2 {
  box-shadow: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: 700;
}

.c2:hover,
.c2:focus {
  border-bottom: 1px solid undefined;
  outline: none;
}

.c0 {
  margin: 0 0 60px 0;
  font-weight: 400;
}

.c1 {
  margin: 20px 0 0 0;
}

.c3 {
  line-height: 1.5;
  font-size: 19px;
  margin-top: 10px;
}

.c3 a {
  text-decoration: none;
  border-bottom: 1px solid;
}

.c3 a:hover,
.c3 a:focus {
  border-bottom: 1px solid;
  outline: none;
}

.c3 code {
  background: 40;
  padding: 2px 6px;
  border-radius: 10px;
  font-size: 16px;
}

.c3 blockquote {
  border-left: 5px solid;
  padding-left: 20px;
  margin-left: 0;
}

.c3 img {
  box-shadow: none!important;
}

.c3 h3 {
  font-size: 28px;
}

.c3 table.uses-table {
  width: 100%;
  margin: 40px 0;
}

.c3 table.uses-table td {
  text-align: center;
}

<body>
  <div>
    <article
      class="c0"
      data-testid="article-item"
    >
      <header
        class="c1"
      >
        <a
          class="c2"
          data-testid="article-header-title"
          href="/awesome-article"
          language="pt"
        >
          Awesome Article
        </a>
      </header>
      <section
        class="c3"
      >
        Article as a Children
      </section>
    </article>
  </div>
</body>
`;

exports[`<Article /> renders the article details 1`] = `
.c2 {
  box-shadow: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: 700;
}

.c2:hover,
.c2:focus {
  border-bottom: 1px solid undefined;
  outline: none;
}

.c4 {
  display: flex;
  flex-wrap: wrap;
  padding: 0;
  margin: 0;
}

.c5 {
  list-style: none;
  padding: 0 10px 0 0;
}

.c6 {
  text-decoration: none;
  box-shadow: none;
  font-size: 16px;
  font-weight: 700;
  text-transform: lowercase;
}

.c6:hover,
.c6:focus {
  border-bottom: 1px solid;
  outline: none;
}

.c0 {
  margin: 0 0 60px 0;
  font-weight: 400;
}

.c1 {
  margin: 20px 0 0 0;
}

.c3 {
  font-size: 19px;
  margin: 10px 0;
}

<body>
  <div>
    <article
      class="c0"
      data-testid="article-item"
    >
      <header
        class="c1"
      >
        <a
          class="c2"
          data-testid="article-header-title"
          href="/awesome-article"
          language="en"
        >
          Awesome Article
        </a>
        <p
          class="c3"
        >
          18/01/2020
           
          · 5 article.minutes article.ofReading
           · 
          article.languagePrefix languages.en
        </p>
        <ul
          class="c4"
          data-testid="article-header-tags"
        >
          <li
            class="c5"
            data-testid="article-header-tag"
          >
            <a
              class="c6"
              href="/tags/jest"
            >
              #jest
            </a>
          </li>
          <li
            class="c5"
            data-testid="article-header-tag"
          >
            <a
              class="c6"
              href="/tags/testing-library"
            >
              #testing-library
            </a>
          </li>
        </ul>
      </header>
    </article>
  </div>
</body>
`;
